Returnontransfer.class.php

```diff
@@ -40,39 +50,76 @@ public function restore($backup)
 	//
 	// This handles any data passed to this module before the page is rendered.
 	public function doConfigPageInit($page) {
-                error_log($_REQUEST['actiondd']);
 		//Handle form submissions
 		switch ($_REQUEST['action']) {
-		case 'save':
-                    $dbh = \FreePBX::Database();
-                    foreach (['timeout','prefix','alertinfo'] as $keyword) {
-                        $sql = "REPLACE INTO returnontransfer (keyword,value) VALUES (?,?)";
-                        $stmt = $dbh->prepare($sql);
-                        $stmt->execute(array($keyword,$_REQUEST[$keyword]));
-                    }
-                    $sql = array();
-                    if ($_REQUEST['enabled']==1) {
-                        $sql[] = "REPLACE INTO `returnontransfer` (keyword,value) VALUES ('enabled','true')";
-                        $sql[] = "UPDATE `freepbx_settings` SET `value` = 'blindxfer_ringback' WHERE `keyword` = 'TRANSFER_CONTEXT'";
-                    } else {
-                        $sql[] = "REPLACE INTO `returnontransfer` (keyword,value) VALUES ('enabled','false')";
-                        $sql[] = "UPDATE `freepbx_settings` SET `value` = 'from-internal-xfer' WHERE `keyword` = 'TRANSFER_CONTEXT'";
-                    }
-                    foreach ($sql as $query) {
-                        $stmt = $dbh->prepare($query);
-                        $stmt->execute(array());
-                    }
-                    needreload();
-		    break;
+			case 'save':
+			foreach (['timeout','prefix','alertinfo'] as $keyword) {
+				$this->setConfig($keyword,$_REQUEST[$keyword]);
+			}
+			$sql = array();
+			if ($_REQUEST['enabled']==1) {
+				$this->setConfig('enabled',true);
+				$this->FreePBX->Config->update('TRANSFER_CONTEXT','blindxfer_ringback');
+			} else {
+				$this->setConfig('enabled',false);
+				$this->FreePBX->Config->reset_conf_settings(array('TRANSFER_CONTEXT'),true);
+			}
+			needreload();
+			break;
 		}
 	}
 
+	// We want to do dialplan stuff.
+	public static function myDialplanHooks()
+	{
+		return 900; //at the very last instance
+	}
+
+	public function doDialplanHook(&$ext, $engine, $priority)
+	{
+		$settings = $this->getAll();
+		$timeout = !empty($settings['timeout']) ? $settings['timeout'] : 15;
+		$prefix = !empty($settings['prefix']) ? empty($settings['prefix']) : 'RT: ${xfer_exten} ${CALLERID(name)}';
```

Comment:
I think this should be
```suggestion
		$prefix = !empty($settings['prefix']) ? $settings['prefix'] : 'RT: ${xfer_exten} ${CALLERID(name)}';
```

Reply:
heh yeah! I messed that up!